### PR TITLE
Feat: Added support for deleting Tags and Tags category.

### LINKED ISSUE #3013: Delete Tags and Tag Categories
Hi!

I have figured out how to add new tags and categories to use custom tags. But I would also like to have the opportunity to remove the tags and categories, so that I can clean up the space. I saw that you did not have the opportunity  to remove them in the localhost server, but do you have any functions that I can use, or ideas to make this work?

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

I worked on added support for deleting tags and tags categories. also added a unit test for the delete functionality.
Closes #3013 
#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :
image
image
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ui

### CODE REVIEW

**Comment by harshach** [comment]
@ShaileshParmar11 thanks for adding the tests and making it near 90%. Looks like tagsAPI.ts doesn't have as much coverage
image
can you spend sometime and increase that coverage as well

**Comment by ShaileshParmar11** [comment]
> @ShaileshParmar11 thanks for adding the tests and making it near 90%. Looks like tagsAPI.ts doesn't have as much coverage image can you spend sometime and increase that coverage as well

Yes, @harshach I will look into it.

File: openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx [hunk 1]
```diff
@@ -66,6 +69,19 @@ import {
...
 
+type DeleteTagsType = {
+  data:
+    | {
+        id: string;
+        name: string;
+        categoryName?: string;
+        isCategory: boolean;
+        status?: LoadingState;
+      }
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx:80]: Yes, I will create a type interface for this.

[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx]: Hey, @vivekratnavel I have updated the code.

File: openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx [hunk 2]
```diff
@@ -632,6 +807,20 @@ const TagsPage = () => {
...
                   />
                 )}
+                {deleteTags.state && (
+                  <ConfirmationModal
+                    bodyText={`Are you sure you want to delete the tag ${
+                      deleteTags.data?.isCategory ? 'category' : ''
+                    } "${deleteTags.data?.name}"?`}
+                    cancelText="Cancel"
+                    confirmText="Confirm"
+                    header="Delete Team"
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx:817]:
```suggestion
                    header="Delete Tag"
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx:817]: @Sachin-chaurasiya nice catch on this, thanks for highlighting.

**Comment by ShaileshParmar11** [comment]
> @ShaileshParmar11 thanks for adding the tests and making it near 90%. Looks like tagsAPI.ts doesn't have as much coverage image can you spend sometime and increase that coverage as well

@harshach I have added a unit test for tagsAPI.ts

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by pmbrull** [comment]
@ShaileshParmar11 will this also be applied to the glossary?

**Comment by ShaileshParmar11** [comment]
> @ShaileshParmar11 will this also be applied to the glossary?

This patch contains support only for the tags page.

**Pull request merged**